Sourcery/Parsing/Parser.swift

```diff
@@ -453,18 +456,57 @@ extension Parser {
 
         return annotations
     }
+    
+    fileprivate func parseTypealiases(from source: [String: SourceKitRepresentable], containingType: Type?, processed: [[String: SourceKitRepresentable]]) -> [Typealias] {
+        var contents = self.contents
+        
+        //replace all processed substructures with whitespaces so that we don't process their typealiases again
+        for substructure in processed {
```

Comment:
That's actually already happening as in `walkType` we create new `processed` array again (line 189). This serves to track processed inner types of each other type. We also keep one separate `processed` for global level types.